src/TSMapEditor/UI/Windows/TriggersWindow.cs

```diff
@@ -919,14 +948,27 @@ private void EventWindowDarkeningPanel_Hidden(object sender, EventArgs e)
             if (editedTrigger == null || lbEvents.SelectedItem == null || selectEventWindow.SelectedObject == null)
                 return;
 
+            TriggerEventType triggerEventType = selectEventWindow.SelectedObject;
             TriggerCondition condition = editedTrigger.Conditions[lbEvents.SelectedIndex];
             condition.ConditionIndex = selectEventWindow.SelectedObject.ID;
 
-            if (selectEventWindow.SelectedObject.P1Type == TriggerParamType.Unused)
-                condition.Parameter1 = 0;
+            for (int i = 0; i < TriggerEventType.MAX_PARAM_COUNT; i++)
+            {
+                if ((int)triggerEventType.Parameters[i].TriggerParamType < 0)
+                {
+                    condition.Parameters[i] = Math.Abs((int)triggerEventType.Parameters[i].TriggerParamType).ToString(CultureInfo.InvariantCulture);
+                    continue;
+                }
 
-            if (selectEventWindow.SelectedObject.P2Type == TriggerParamType.Unused)
-                condition.Parameter2 = 0;
+                if (triggerEventType.Parameters[i].TriggerParamType == TriggerParamType.Unused)
+                {
+                    if (i == 2)
```

Comment:
Just for some added context, I can see this magic number being confusing otherwise if one looks at this code years later (or without having knowledge on how the trigger events work).

```suggestion
                    // P3 needs to be empty instead of 0 if it's unused
                    if (i == TriggerCondition.MAX_PARAM_COUNT - 1) 
```